include/revng/ADT/TrackingContainer.h

```diff
@@ -222,7 +252,8 @@ class TrackingContainer {
   size_type count(const key_type &Key) const {
     auto Quantity = Content.count(Key);
     if (Quantity == 0) {
-      NonExisting.back().insert(Key);
+      if (IsTracking)
```

Comment:
Hoist out of `if (Quantity == 0) {`

Reply:
not sure what you mean, i can't just hoist it out, the semantic is not the same

Reply:
I'm suggesting

```c++
    if (TrackingIsActive) {
      if (Quantity == 0) {
       NonExisting.back().insert(Key);
      } else {
        markExistingKey(Key);
      }
    }
```

Reply:
ahh right, yeah make sense